src/commonMain/kotlin/komem/litmus/runners/LitmusTestRunnerUtil.kt

```diff
@@ -0,0 +1,74 @@
+package komem.litmus.runners
+
+import komem.litmus.*
+import kotlin.time.Duration
+import kotlin.time.TimeSource
+
+fun LitmusTestRunner.runTest(
+    timeLimit: Duration,
+    params: RunningParams,
+    testProducer: () -> LitmusTest,
+): LitmusResult {
+    val results = mutableListOf<LitmusOutcomeInfo>()
+    val start = TimeSource.Monotonic.markNow()
+    while (start.elapsedNow() < timeLimit) {
+        val outcomes = runTest(params, testProducer)
+        val outcomeSetup = testProducer().getOutcomeSetup()
+        results.addAll(outcomes.groupToInfo(outcomeSetup))
+    }
+    return results.mergeOutcomes()
+}
+
+/*
+ Note: interprets AffinityMap as a sequence of smaller maps
+ Example: for a map [ [0], [1], [2], [3] ],a test with 2 threads, and 2 instances, the
+ first instance will have a [ [0], [1] ] map and the second one will have [ [2], [3] ].
+ */
+fun LitmusTestRunner.runTestParallel(
+    params: RunningParams,
+    instances: Int,
```

Comment:
Similarly, as you make `timeLimit` the first argument in the previous override, it makes sense to make `instances` first argument here too. 

In general, it is a good practice to keep the same order of arguments in all overloads (and do not insert new arguments in-between).